src/pages/home/report/EmojiPickerMenu.js

```diff
@@ -0,0 +1,122 @@
+import React, {Component} from 'react';
+import {View, FlatList, Text} from 'react-native';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import styles from '../../../styles/styles';
+import themeColors from '../../../styles/themes/default';
+import emojis from '../../../../assets/emojis';
+import EmojiPickerMenuItem from './EmojiPickerMenuItem';
+import TextInputFocusable from '../../../components/TextInputFocusable';
+
+const propTypes = {
+    // Controls the visibility of this component.
+    isVisible: PropTypes.bool,
+
+    // Function to add the selected emoji to the main compose text input
+    addEmojiToTextBox: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+    isVisible: false,
+};
+
+class EmojiPickerMenu extends Component {
+    constructor(props) {
+        super(props);
+
+        // Ref for the emoji search input
+        this.searchInput = undefined;
+
+        this.filterEmojis = _.debounce(this.filterEmojis.bind(this), 500, false);
+        this.renderItem = this.renderItem.bind(this);
+
+        this.state = {
+            filteredEmojis: emojis,
+
+            // This is the indices of each category of emojis
+            // The positions are static, and are calculated as index/numColumns (8 in our case)
```

Comment:
That's correct

Reply:
Hmm ok, not really ideal. But maybe at least use a constant for this array so we don't have two arrays of the same weird numbers defined in separate spots?

Reply:
Would be nice if there was some way to really make this clear at the top of the file with a big warning at a minimum since if any new emojis are added things will break.

Reply:
Updated to add a warning and make it a constant